packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -138,6 +140,58 @@ export class Rule extends Resource implements IRule {
 
     const roleArn = targetProps.role ? targetProps.role.roleArn : undefined;
 
+    if (targetProps.target) {
+      const targetStack = Stack.of(targetProps.target);
+      const sourceStack = Stack.of(this);
+      const sourceAccount = sourceStack.account;
+      if (targetStack.account !== sourceAccount) {
```

Comment:
what should we do if either of these is `AWS::AccountId`? I guess if both are tokens, it's fine, but if only one, we shouldn't regard this as a cross-account and fail.

Reply:
Done.